extensions/amp-form/0.1/amp-form.js

```diff
@@ -0,0 +1,150 @@
+
+    /** @const @private {string} */
+    this.action_ = user.assert(this.form_.getAttribute('action'),
+        'amp-form requires action attribute %s', this.form_);
+
+    this.installSubmitHandler_();
+  }
+
+  /** @private */
+  installSubmitHandler_() {
+    this.form_.addEventListener('submit', e => {
+      // Safari does not trigger validation check on submission, hence we
+      // trigger it manually. In other browsers this would never execute since
+      // the submit event wouldn't be fired if the form is invalid.
+      if (!this.form_.checkValidity()) {
+        this.form_.classList.remove('valid');
+        this.form_.classList.add('invalid');
+        // TODO: This would prevent submission but Safari won't display any messaging.
+        // Validation messages are available on fields provided by browser using
+        // the .validity object. We need to provide a bubble UI to display
+        // these messages.
+        e.preventDefault();
+        return false;
+      }
+
+      this.form_.classList.add('valid');
+      if (this.form_.classList.contains('invalid')) {
+        this.form_.classList.remove('invalid');
+        this.form_.classList.add('was-invalid');
+      }
+
+      this.xhr_.fetchJson(this.action_, {
+        body: new FormData(this.form_),
+        method: this.method_,
+      }).then(response => this.successHandler_(response))
+          .catch(error => this.errorHandler_(error));
+
+      e.preventDefault();
+      return false;
+    });
+  }
+
+  /**
+   * @param {!Object} response.
+   * @private
+   */
+  successHandler_(response) {
+    this.form_.classList.remove('error');
+    this.form_.classList.add('success');
+    this.renderTemplate_('success', response);
+  }
+
+  /**
+   * @param {!Object} error.
+   * @private
+   */
+  errorHandler_(error) {
+    this.form_.classList.remove('success');
+    this.form_.classList.add('error');
+    this.renderTemplate_('error', error);
+  }
+
+  /**
+   * Renders the template with the given name if one exists.
+   * @param {string} name
+   * @param {!Object} data
+   * @private
+   */
+  renderTemplate_(name, data) {
+    const template = childElement(this.form_,
+        el => el.tagName == 'TEMPLATE' && el.hasAttribute(name));
+    if (template) {
+      // Remove previously rendered messages to clear for new message.
+      const messageElement = this.form_.querySelector('.amp-form-messages');
+      if (messageElement) {
+        removeElement(messageElement);
+      }
+      this.templates_.renderTemplate(template, data).then(element => {
+        element.classList.add('amp-form-messages');
+        this.form_.appendChild(element);
+      });
+    }
+  }
+
+}
+
+function installSubmissionHandlers(win) {
+  onDocumentReady(win.document, () => {
+    toArray(win.document.forms).forEach(form => {
+      new AmpForm(form, win);
+    });
+  });
+}
+
+export function installAmpForms(win) {
+  return getService(win, 'amp-forms', () => {
```

Comment:
Let's change to `amp-form`. Or whichever way, but we must always install the service with the name of the extension. We can have two services if we want to keep `amp-forms` name.